src/overloads/arrays.jl

```diff
@@ -0,0 +1,208 @@
+"""
+    second_order_or(tracers)
+
+Compute the most conservative elementwise OR of tracer sparsity patterns,
+including second-order interactions to update the `hessian` field of `HessianTracer`.
+
+This is functionally equivalent to:
+```julia
+reduce(^, tracers)
+```
+"""
+function second_order_or(ts::AbstractArray{T}) where {T<:AbstractTracer}
+    # TODO: improve performance
+    return reduce(second_order_or, ts; init=myempty(T))
+end
+
+function second_order_or(a::T, b::T) where {T<:ConnectivityTracer}
+    return connectivity_tracer_2_to_1(a, b, false, false)
+end
+function second_order_or(a::T, b::T) where {T<:GradientTracer}
+    return gradient_tracer_2_to_1(a, b, false, false)
+end
+function second_order_or(a::T, b::T) where {T<:HessianTracer}
+    return hessian_tracer_2_to_1(a, b, false, false, false, false, false)
+end
+
+"""
+    first_order_or(tracers)
+
+Compute the most conservative elementwise OR of tracer sparsity patterns,
+excluding second-order interactions of `HessianTracer`.
+
+This is functionally equivalent to:
+```julia
+reduce(+, tracers)
+```
+"""
+function first_order_or(ts::AbstractArray{T}) where {T<:AbstractTracer}
+    # TODO: improve performance
+    return reduce(first_order_or, ts; init=myempty(T))
+end
+function first_order_or(a::T, b::T) where {T<:ConnectivityTracer}
+    return connectivity_tracer_2_to_1(a, b, false, false)
+end
+function first_order_or(a::T, b::T) where {T<:GradientTracer}
+    return gradient_tracer_2_to_1(a, b, false, false)
+end
+function first_order_or(a::T, b::T) where {T<:HessianTracer}
+    return hessian_tracer_2_to_1(a, b, false, true, false, true, true)
+end
+
+#===========#
+# Utilities #
+#===========#
+
+function split_dual_array(A::AbstractArray{D}) where {D<:Dual}
+    primals = getproperty.(A, :primal)
+    tracers = getproperty.(A, :tracer)
+    return primals, tracers
+end
+function split_dual_array(A::SparseArrays.SparseMatrixCSC{D}) where {D<:Dual}
+    A = Matrix(A)
+    primals = getproperty.(A, :primal)
+    tracers = getproperty.(A, :tracer)
+    return sparse(primals), sparse(tracers)
+end
```

Comment:
why is this necessary?

Reply:
To split a sparse matrix of duals into 
1. a sparse matrix of primals 
2. a sparse matrix of tracers

`split_dual_array` is called in `det`, `logdet` and `logabsdet`.